test/src/test/java/org/corfudb/AbstractCorfuTest.java

```diff
@@ -102,6 +104,12 @@ public void cleanupScheduledThreads() {
         scheduledThreads.clear();
     }
 
+    @After
+    public void cleanPerTestTempDir() {
+        deleteFolder(new File(PARAMETERS.TEST_TEMP_DIR),
+                false);
+    }
+
```

Comment:
Ok, there's a good reason for this, and we probably need to make this clearer to programmers (maybe call it getNewTempDir()): 

Every time you call getTempDir(), you get a new temp directory. I can envision a case where you might want this (maybe you have two servers, for example, each which needs its own directory). This behavior, however could lead to bugs, if you thought that getting a temp directory would give you the same directory each time (for example, in @Maithem 's test, the second call resulted in a new temporary directory)...

Hence the move to creating a per-test temporary directory which remains constant throughout the entire unit test execution. I'll see if I can refactor it better..